python/docs/source/development/ps_design.rst

```diff
@@ -1,85 +0,0 @@
-=================
-Design Principles
-=================
-
-.. currentmodule:: pyspark.pandas
-
-This section outlines design principles guiding the pandas API on Spark.
-
-Be Pythonic
------------
-
-Pandas API on Spark targets Python data scientists. We want to stick to the convention that users are already familiar with as much as possible. Here are some examples:
-
-- Function names and parameters use snake_case, rather than CamelCase. This is different from PySpark's design. For example, pandas API on Spark has `to_pandas()`, whereas PySpark has `toPandas()` for converting a DataFrame into a pandas DataFrame. In limited cases, to maintain compatibility with Spark, we also provide Spark's variant as an alias.
-
-- Pandas API on Spark respects to the largest extent the conventions of the Python numerical ecosystem, and allows the use of NumPy types, etc. that can be supported by Spark.
-
-- pandas-on-Spark docs' style and infrastructure simply follow rest of the PyData projects'.
-
-Unify small data (pandas) API and big data (Spark) API, but pandas first
```

Comment:
This is removed. Now pandas-on-Spark specific APIs are placed under `DataFrame.pandas_on_spark`. PySpark APIs are placed under `DataFrame.spark` for now. `DataFrame` mainly only hold pandas APIs only.